### PR TITLE
Add throwOnStdErr option

### LINKED ISSUE #20: 40d9f0efcae1ab61302cc3b90a75385c31966578 broke linter-js-standard
https://github.com/AtomLinter/atom-linter/commit/40d9f0efcae1ab61302cc3b90a75385c31966578 broke linter-js-standard. See https://github.com/ricardofbarros/linter-js-standard/issues/30

### PR DESCRIPTION
**Pull request by steelbrain**
Fixes #20

### CODE REVIEW

**Comment by AsaAyers** [comment]
Tests?

**Comment by steelbrain** [comment]
Yep, writing them.

**Comment by steelbrain** [comment]
Added tests, all green locally.

File: lib/helpers.coffee [hunk 1]
```diff
@@ -32,7 +33,7 @@ module.exports = Helpers =
           resolve(data.stderr.join(''))
       if isNodeExecutable
         options.env ?= {}
-        for prop,value of process.env
+        for prop, value of process.env
```
Comment (iam4x) [inline lib/helpers.coffee]: :+1:  :100:

File: spec/helper-spec.coffee [hunk 2]
```diff
@@ -35,6 +36,24 @@ describe 'linter helpers', ->
       waitsForPromise ->
         helpers.exec(__dirname + '/fixtures/stderr.sh', []).catch (message) ->
           expect(message).toBe("STDERR\n")
+
+    describe 'throwOnStdErr option', ->
+      it 'throws unexpected error when set to true', ->
+        gotError = false
+        waitsForPromise ->
+          helpers.exec(__dirname + '/fixtures/stderr.sh', [], throwOnStdErr: true).catch( (message) ->
```
Comment (iam4x) [inline spec/helper-spec.coffee]: `"#{__dirname}/fixtures/stderr.sh"

Let's use Coffee features!

File: spec/helper-spec.coffee [hunk 3]
```diff
@@ -35,6 +36,24 @@ describe 'linter helpers', ->
...
+            gotError = true
+            expect(message).toBe("STDERR\n")
+          ).then ->
+            expect(gotError).toBe(true)
+      it 'suppresses unexpected errors when set to false', ->
+        gotError = false
+        waitsForPromise ->
+          helpers.exec(__dirname + '/fixtures/stderr.sh', [], throwOnStdErr: false).catch(->
```
Comment (iam4x) [inline spec/helper-spec.coffee]: Same as above ^

**Comment by steelbrain** [comment]
@iam4x done

**Pull request merged**